purchase_request_to_rfq/models/purchase_order.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+
+from openerp import _, api, exceptions, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.model
+    def _purchase_request_confirm_message_content(self, po, request,
+                                                  request_dict):
+        if not request_dict:
+            request_dict = {}
+        title = _('Order confirmation %s for your Request %s') % (
+            po.name, request.name)
+        message = '<h3>%s</h3><ul>' % title
+        message += _('The following requested items from Purchase Request %s '
+                     'have now been confirmed in Purchase Order %s:') % (
+            request.name, po.name)
+
+        for line in request_dict.values():
+            message += _(
+                '<li><b>%s</b>: Ordered quantity %s %s, Planned date %s</li>'
+            ) % (line['name'],
+                 line['product_qty'],
+                 line['product_uom'],
+                 line['date_planned'],
+                 )
+        message += '</ul>'
+        return message
+
+    @api.multi
+    def _purchase_request_confirm_message(self):
+        request_obj = self.env['purchase.request']
+        for po in self:
+            requests_dict = {}
+            for line in po.order_line:
+                for request_line in line.sudo().purchase_request_lines:
+                    request_id = request_line.request_id.id
+                    if request_id not in requests_dict:
+                        requests_dict[request_id] = {}
+                    date_planned = "%s" % line.date_planned
+                    data = {
+                        'name': request_line.name,
+                        'product_qty': line.product_qty,
+                        'product_uom': line.product_uom.name,
+                        'date_planned': date_planned,
+                    }
+                    requests_dict[request_id][request_line.id] = data
+            for request_id in requests_dict:
+                request = request_obj.sudo().browse(request_id)
+                message = self._purchase_request_confirm_message_content(
+                    po, request, requests_dict[request_id])
+                request.message_post(body=message, subtype='mail.mt_comment')
+        return True
+
+    @api.multi
+    def _purchase_request_line_check(self):
+        for po in self:
```

Comment:
Becuase you need to tell the user what purchase request has already been completed, and with this approach you cannot know what request line fails the validation
                        raise exceptions.Warning(
                            _('Purchase Request %s has already '
                              'been completed') % request_line.request_id.name)